src/Operators/remapping.jl

```diff
@@ -20,8 +39,15 @@ A remapping operator from the `source` space to the `target` space.
 See [Ullrich2015](@cite) eqs. 3 and 4.
 """
 function LinearRemap(target::AbstractSpace, source::AbstractSpace)
-    R = linear_remap_op(target, source)
-    LinearRemap(target, source, R)
+    if typeof(target) == typeof(source) && (
+        parent(Spaces.local_geometry_data(target)) ==
+        parent(Spaces.local_geometry_data(source))
+    )
```

Comment:
For posterity (@simonbyrne and I chatted): currently `==` defaults to `===` for Spaces. This is not what we want to check if we are interested in avoiding "same space different instance" errors or when mapping between spaces that are the same, even if different instances. This check should be same as the one we use in the Field broadcast error handling since an `IdentityRemap` application amounts to a simply broadcast.

This does raise the question of whether we'd want to extend `Base.==` for Spaces, however, and what that would look like.